state/execution.go

```diff
@@ -163,6 +202,20 @@ func (blockExec *BlockExecutor) ProcessProposal(
 	block *types.Block,
 	state State,
 ) (bool, error) {
+	txs := block.Data.Txs.ToSliceOfBytes()
+	if len(txs) > 0 {
+		res, err := blockExec.proxyApp.ValidateOracleVotes(context.TODO(), &abci.RequestValidateOracleVotes{OracleTx: txs[0]})
```

Comment:
why are we using a random context?

Reply:
following the convention as processProposal also runs with context.TODO(), i believe the actual context used is set on cosmos-sdk side, when we initialise the hook (which is the processProposalCtx)

ref: https://github.com/Switcheo/cosmos-sdk/blob/feat/gossip-oracle/baseapp/abci.go#L994